### ISSUE TITLE
list[EventList]() is broken

### ISSUE THREAD

**Issue by alanbur** [doriordan/skuber#149]
Calling:

```k8s.list[EventList]()```

results in:

```
17:12:28.243 INFO  [skuber.api] [ { reqId=e8dcc18c-0412-46f8-83f7-c073ac44c8e7} } - about to send HTTP request: GET https://XXXX:6443/api/v1/namespaces/alan/events]
sleep
17:12:28.586 INFO  [skuber.api] [ { reqId=e8dcc18c-0412-46f8-83f7-c073ac44c8e7} } - received response with HTTP status 200]
:
failed to get event list: akka.http.scaladsl.server.RejectionError
```

From poking around in the code this looks like it might be a JSON deserialization error, but I haven't located exactly where. This is a critical bug due to the way the K8s watch API has been implemented - by default watching events delivers all events that K8s has in its history, **not** all events from the time the watch is created onwards, so you get 'old' events. To avoid that you need to list the events and use the resourceVersion from the returned list as the starting point via the sinceResourceVersion parameter to the watch. This can be confirmed by looking at the output of `kubectl get events --watch-only -v9`

**Comment by alanbur** [comment]
The issue might be that the same Event object is used for both 'get' and 'list' operations, but the JSON payload for Events is different between the two, namely the 'get' JSON has two fields, `apiVersion` and `kind`, that are missing from the 'list' JSON:

From 'list':
```
{
  "count": 1,
  "eventTime": null,
  "firstTimestamp": "2018-05-16T19:01:12Z",
  "involvedObject": {
    "apiVersion": "v1",
    "kind": "Pod",
    "name": "test-1-1",
    "namespace": "alan",
    "resourceVersion": "4556350",
    "uid": "8046a28c-593b-11e8-8f04-6624a4fcdae3"
  },
  "lastTimestamp": "2018-05-16T19:01:12Z",
  "message": "Successfully assigned test-1-1 to foo",
  "metadata": {
    "creationTimestamp": "2018-05-16T19:01:12Z",
    "name": "test-1-1.152f352e7d6d4ab9",
    "namespace": "alan",
    "resourceVersion": "4556352",
    "selfLink": "/api/v1/namespaces/alan/events/test-1-1.152f352e7d6d4ab9",
    "uid": "8047a198-593b-11e8-8f04-6624a4fcdae3"
  },
  "reason": "Scheduled",
  "reportingComponent": "",
  "reportingInstance": "",
  "source": {
    "component": "default-scheduler"
  },
  "type": "Normal"
}
```

From 'get':

```
{
  "apiVersion": "v1",
  "count": 1,
  "eventTime": null,
  "firstTimestamp": "2018-05-16T19:01:12Z",
  "involvedObject": {
    "apiVersion": "v1",
    "kind": "Pod",
    "name": "test-1-1",
    "namespace": "alan",
    "resourceVersion": "4556350",
    "uid": "8046a28c-593b-11e8-8f04-6624a4fcdae3"
  },
  "kind": "Event",
  "lastTimestamp": "2018-05-16T19:01:12Z",
  "message": "Successfully assigned test-1-1 to foo",
  "metadata": {
    "creationTimestamp": "2018-05-16T19:01:12Z",
    "name": "test-1-1.152f352e7d6d4ab9",
    "namespace": "alan",
    "resourceVersion": "4556352",
    "selfLink": "/api/v1/namespaces/alan/events/test-1-1.152f352e7d6d4ab9",
    "uid": "8047a198-593b-11e8-8f04-6624a4fcdae3"
  },
  "reason": "Scheduled",
  "reportingComponent": "",
  "reportingInstance": "",
  "source": {
    "component": "default-scheduler"
  },
  "type": "Normal"
}
```

**Comment by doriordan** [comment]
You are right in that the apiVersion and kind missing were the issues, these are optional fields (although usually present) but the json formatters for Event and a few other types were treating them as mandatory. Fixed by #152.

**Comment by tgpfeiffer** [comment]
With `"io.skuber" %% "skuber" % "2.0.9"` this is still happening for me with `list[NodeList]`:

```
default-akka.actor.default-dispatcher-14] INFO  skuber.api - [ { reqId=d6cff128-abe8-4213-adab-af405490bc0d} } - received response with HTTP status 200]
e => JsError(List((/items(0)/status/images(0)/sizeBytes,List(JsonValidationError(List(error.expected.int),WrappedArray()))), (/items(0)/status/images(1)/sizeBytes,List(JsonValidationError(List(error.expected.int),WrappedArray()))), (/items(0)/status/images(2)/sizeBytes,List(JsonValidationError(List(error.expected.int),WrappedArray()))), (/items(0)/status/images(3)/sizeBytes,List(JsonValidationError(List(error.expected.int),WrappedArray()))), ...
```

I guess that may be the same root cause as for `EventList`? Should I try and send a fix or is there some other workaround?
